### PR TITLE
Cqc provider clean script setup

### PR DESCRIPTION
**Pull request by luke-a-williams**
# Description
All of the code plus supporting code from the pairing task.
Some extras:
Addition of ipykernal to dev dependencies for local testing

# How to test

# Developer checklist
- [x] Unit test
- [x] Linked to Trello ticket
- [x] Documentation up to date

### CODE REVIEW

**Review by garypolzin — CHANGES_REQUESTED** [review]
Will be worth working with Marion to iron out some of the differences between this PR and https://github.com/NMDSdevopsServiceAdm/DataEngineering/pull/236 to maintain consistency and also define how we're going to progress with the rest of the project.
Particularly in relation to how columns names and partition keys etc are stored/called

Some other minor comments with unused imports but in general looks good and enjoyed pairing on it!

File: jobs/clean_cqc_provider_data.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import sys
+
+import pyspark.sql.dataframe
+
+from utils import utils
+
+
+def clean_cqc_provider_df(df_to_clean) -> pyspark.sql.DataFrame:
```
Comment (james-francis-MT) [inline jobs/clean_cqc_provider_data.py:8]: Is it worth making this function beforehand? Seems like it is pushing future work in a direction

Comment (luke-a-williams) [reply jobs/clean_cqc_provider_data.py]: Agreed that we would worry about this in the future tickets that are specific to cleaning.

**Review by MarionJHolloway — CHANGES_REQUESTED** [review]

File: tests/unit/test_clean_cqc_provider_data.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    partition_keys = ["year", "month", "day", "import_date"]
+
+    def setUp(self) -> None:
+        self.spark = utils.get_spark()
+        self.test_cqc_providers_parquet = self.spark.createDataFrame(
+            Data.sample_rows_full, schema=Schemas.full_parquet_schema
+        )
+
+    def test_clean_cqc_provider_df(self):
+        # Test returned df is the same as the one passed in
```
[the remark this answers is not in the document]
Comment (luke-a-williams) [reply tests/unit/test_clean_cqc_provider_data.py]: method has been removed anyway

File: tests/test_helpers.py [hunk 3]
```diff
@@ -1,8 +1,16 @@
...
 
 
 def remove_file_path(path):
     try:
         shutil.rmtree(path)
     except OSError:
         pass
+
+
+def create_spark_mock():
```
Comment (james-francis-MT) [inline tests/test_helpers.py:12]: Not being used?

**Review by garypolzin — APPROVED** [review]

**Review by MarionJHolloway — APPROVED** [review]
Changes agreed among the team

File: jobs/clean_cqc_provider_data.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+from utils import utils
+
+
+def main(cqc_source: str, cleaned_cqc_destination: str):
+    cqc_provider_df = utils.read_from_parquet(cqc_source)
+    utils.write_to_parquet(
+        cqc_provider_df,
+        cleaned_cqc_destination,
+        append=True,
+        partitionKeys=["year", "month", "day", "import_date"],
```
Comment (garypolzin) [inline jobs/clean_cqc_provider_data.py:14]: Just spotted that the partitionKeys haven't been updated to the Key. approach

**Review by garypolzin — CHANGES_REQUESTED** [review]

**Review by garypolzin — APPROVED** [review]

**Review by james-francis-MT — APPROVED** [review]

**Pull request merged**